### DISCUSSION TITLE
🔗 Advanced Forms Capabilities

### DISCUSSION THREAD

**Discussion by orofri** [elementor/elementor#21783] [Feature Request]
### Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Since "Gravity Forms" and "Hello Elementor" theme doesn't work so well together and get some conflicts, it would be nice to enhance the Elementor forms widget in a way that we won't need to rely on 3rd party plugins like "Gravity forms".

### Describe the solution you'd like

It would be nice to have the features below:

1. Custom registration form: Ability to register new users using Elementor forms. For this, we will need:
- [ ] A username field that would validate that there's no existing user with the same name.
- [ ] Add options for password validation and password weakness check/requirements to the existing password field.
- [ ] (optional)Ability to custom fields to the user such as birthday, phone number, area of interest etc.
- [ ] Ability to register the user on submission.

2. Surveys: it would be nice to be able to have a survey field where the user will have a question and 5 options that he can choose from, for example, "Strongly disagree" to "Strongly Agree" in a nice visual way like most surveys. 

3. (optional)Digital signature: if I want to ask users to sign on the "terms and conditions" of the company it would be good to have this option.

4. Anti-Spam: OTP - sms verification(using AWS SNS) - it will help to avoid human spammers, especially in businesses with a lot of competition - it's very easy to setup AWS account anywhere in the world so I think having this gateway integration would be beneficial for most of the Elementor customers.

Thank you so much!

**Comment by bogdanconstantinzanescu** [comment]
-Besides password validation I would add option to stop storing data from password field when Form Submission action after submit is triggered.
-adding to what orofri wrote, requesting feature for popup when user failed submission and not displaying the failed text under the form because some forms are quite long and people don´t see the failed message and leave, waiting for email

**Comment by BastiaanBosch** [comment]
Having the ability to crate a new-user-registration form, with reCAPTCHA, confirmation link etc..,
without the need of an external 3rd party plugin (eg TTFB, plugin conflicts, and other server loading issues) is LONG overdue.

**Comment by drewclifton** [comment]
Please merge this with duplicate [Improved form widget - Action for Login, Registration And Forget password. #21600](https://github.com/orgs/elementor/discussions/21600)
